sdk/servicebus/Azure.Messaging.ServiceBus/src/Primitives/ServiceBusMessage.cs

```diff
@@ -62,7 +62,15 @@ public ServiceBusMessage(BinaryData body)
         public ServiceBusMessage(ServiceBusReceivedMessage receivedMessage)
         {
             Argument.AssertNotNull(receivedMessage, nameof(receivedMessage));
-            AmqpMessage = new AmqpAnnotatedMessage(new ReadOnlyMemory<byte>[] { receivedMessage.Body });
+
+            if (receivedMessage.AmqpMessage.Body is AmqpDataMessageBody dataBody)
+            {
+                AmqpMessage = new AmqpAnnotatedMessage(dataBody.Data);
+            }
+            else
+            {
+                throw new NotSupportedException($"{receivedMessage.Body.GetType()} is not a supported message body type.");
```

Comment:
Should we add a `.Name` at the end to ensure consistent messaging across platforms?   Pedantically, this shouldn't make a difference, but it may be worth taking explicit control, just in case.